src/allmydata/scripts/create_node.py

```diff
@@ -344,6 +350,54 @@ def create_node(config):
         os.mkdir(basedir)
     write_tac(basedir, "client")
 
+    # if we're doing magic-wormhole stuff, do it now
+    if config['join'] is not None:
+        print >>out, "Opening wormhole with code '{}'".format(config['join'])
+
+        wh = wormhole.create(
+            appid=u"tahoe-lafs.org/lafs",
+            relay_url=u"ws://wormhole.leastauthority.com:4000/v1",
+            reactor=reactor,
+        )
+        code = unicode(config['join'])
+        wh.set_code(code)
+        yield wh.get_welcome()
+        print >>out, "Connected to wormhole server"
+
+        intro = {
+            u"abilities": {
+                "client-v1": {},
+            }
+        }
+        wh.send_message(json.dumps(intro))
+
+        server_intro = yield wh.get_message()
+        server_intro = json.loads(server_intro)
+
+        print >>out, "  received server introduction"
+        assert u'abilities' in server_intro
+        assert u'server-v1' in server_intro['abilities']
+
+        remote_data = yield wh.get_message()
+        print >>out, "  received configuration"
+
+        # XXX do we want to whitelist what config this can set?
+        remote_config = json.loads(remote_data)
+        print >>out, "Encoding: {needed} of {total} shares, on at least {happy} " + \
+            "servers".format(**remote_config)
+        print >>out, "Overriding the following config:"
+        # presume someone doing "--join" doesn't want to provide
+        # storage; if they do they can change the config after (and/or
+        # allow the 'tahoe invite' to specify storage options?).
+        remote_config['no-storage'] = True
+        remote_config['listen'] = 'none'
+        for k, v in remote_config.items():
+            if k not in ['happy', 'needed', 'total', 'introducer']:
+                print >>out, "   --{}={}".format(k, v)
+            config[k] = v
```

Comment:
It *does* map to options, but it turns out the only thing *not* in that list is ``--nickname`` so yeah probably just whitelisting it (er, well explicitly pulling out the ones that are ok) is best.